zikyo-core/shared/src/test/scala/zikyoTest/effectsTest.scala

```diff
@@ -57,5 +69,11 @@ class effectsTest extends ZiKyoTest:
             val handledEffectWhen2 = IOs.run(Options.run(effectWhen))
             assert(handledEffectWhen2.pure == Some(false))
         }
+
+        "tap" in {
+            val effect: Int < IOs = IOs(42).tap(v => assert(42 == v)).as(24)
```

Comment:
I think this should drop the `as(24)` and assert that `handled.pure == 42`. We want to make sure that tap uses the value of the previous effect but does not change it.

Reply:
Done